server/src/main/java/org/elasticsearch/index/fielddata/SortingNumericDocValues.java

```diff
@@ -51,16 +58,40 @@ protected int compare(int i, int j) {
                 return Long.compare(values[i], values[j]);
             }
         };
+        this.circuitBreakerConsumer = circuitBreakerConsumer;
+        // account for initial values size of 1
+        this.circuitBreakerConsumer.accept(Long.BYTES);
     }
 
     /**
      * Set the {@link #docValueCount()} and ensure that the {@link #values} array can
      * store at least that many entries.
      */
     protected final void resize(int newSize) {
+        int oldValuesLength = values.length;
+        long oldValuesSizeInBytes = values.length * Long.BYTES;
+
+        // If array is expected to grow, increment the circuit breaker
+        // to include both the additional bytes used by the grown array
+        // as well as the overhead of keeping both arrays in memory while
+        // copying.
+        if (newSize > values.length) {
```

Comment:
I think it'd be clearer to bail early if the newSize <= values.length rather than relying on the behavior of `ArrayUtils.grow`